master/buildbot/steps/source/p4.py

```diff
@@ -150,11 +151,11 @@ def full(self, _):
         if self.revision:
             if debug_logging:
                 log.msg("P4: full() sync command based on :base:%s changeset:%d", self._getP4BaseForLog(), int(self.revision))
-            yield self._dovccmd(['sync', '%s...@%d' % (self._getP4BaseForCommand(), int(self.revision))], collectStdout=True)
+            yield self._sync(['%s...@%d' % (self._getP4BaseForCommand(), int(self.revision))])
```

Comment:
Hmm... I thought revision would be specified with `#revision`.  From `p4 help revisions`:

```
            #rev - A revision number or one of the following keywords:
            #none - A nonexistent revision (also #0).
            #head - The current head revision (also @now).
            #have - The revision on the current client.
            @change - A change number: the revision as of that change.
            @client - A client name: the revision on the client.
            @label - A label name: the revision in the label.
            @date - A date or date/time: the revision as of that time.
                    Either yyyy/mm/dd or yyyy/mm/dd:hh:mm:ss
                    Note that yyyy/mm/dd means yyyy/mm/dd:00:00:00.
                    To include all events on a day, specify the next day.
```